app/views/forms/_odk_group.xml.erb

```diff
@@ -0,0 +1,20 @@
+<group appearance="field-list">
+  <label><%= qing.first.parent.group_name %></label>
+  <% grid_mode = grid_mode?(qing) %>
+
+  <% if grid_mode %>
+    <%= render('odk_questioning',
+      qing: qing.first,
+      style: style,
+      grid_mode: grid_mode,
+      options: {appearance: 'label'}) %>
+  <% end %>
```

Comment:
I see what you're doing here. And I like it. However, inside odk_questioning a lot of stuff disappears if you're in appearance label mode. See below. I think you can just stuff it right in here and not worrying about calling the partial.